core/src/autogluon/core/utils/early_stopping.py

```diff
@@ -0,0 +1,103 @@
+
+class AbstractES:
+    def update(self, cur_round, is_best=False) -> bool:
+        raise NotImplementedError
+
+    def early_stop(self, cur_round, is_best=False) -> bool:
+        raise NotImplementedError
+
+
+class SimpleES(AbstractES):
+    def __init__(self, stopping_rounds=10):
+        self.stopping_rounds = stopping_rounds
+        self.best_round = 0
+
+    def update(self, cur_round, is_best=False):
+        if is_best:
+            print(f'new best: \t{cur_round}\t{self.best_round}\t{cur_round - self.best_round}\t{(cur_round - self.best_round) / self.stopping_rounds}')
+            self.best_round = cur_round
+        return self.early_stop(cur_round, is_best=is_best)
+
+    def early_stop(self, cur_round, is_best=False):
+        if is_best:
+            return False
+        return cur_round - self.best_round >= self.stopping_rounds
+
+
+# TODO: Add time component
+#  if given a large amount of time and training is fast, should check more rounds before early stopping
+# TODO: Incorporate score, rolling window
+class AdaptiveES(AbstractES):
+    """
+    Implements Adaptive Early Stopping
+
+    stopping_rounds is adaptively adjusted across training instead of being a fixed value.
```

Comment:
Would be good to explain what this does and its advantages since this docstring is the sole description of this.

Reply:
Also would be good to include example of how user can in predictor.fit() specify some of these hyperparameters like `min_rounds`. This seems pretty unclear with the current documentation.

Reply:
Added more documentation and explanation. I think we should separate the `predictor.fit` API documentation from the general low-level components, otherwise the code looks like it is tied to predictor.fit when it is in fact generic. I'd prefer to add support for this functionality to some more models before making the user aware of it.